acpi/src/lib.rs

```diff
@@ -115,170 +138,150 @@ where
     H: AcpiHandler,
 {
     /// Create an `AcpiTables` if you have the physical address of the RSDP.
-    pub unsafe fn from_rsdp(handler: H, rsdp_address: usize) -> Result<AcpiTables<H>, AcpiError> {
-        let rsdp_mapping = unsafe { handler.map_physical_region::<Rsdp>(rsdp_address, mem::size_of::<Rsdp>()) };
+    ///
+    /// ### Safety: Caller must ensure the provided address is valid to read as an RSDP.
+    pub unsafe fn from_rsdp(handler: H, address: usize) -> AcpiResult<Self> {
+        let rsdp_mapping = unsafe { handler.map_physical_region::<Rsdp>(address, mem::size_of::<Rsdp>()) };
         rsdp_mapping.validate().map_err(AcpiError::Rsdp)?;
 
-        Self::from_validated_rsdp(handler, rsdp_mapping)
+        // ### Safety: `RSDP` has been validated.
```

Comment:
Here and elsewhere: having these as headers doesn't really make sense, as they aren't doc-comments, so the Markdown is never going to be rendered no?

Reply:
I've mostly styled them for consistency. I'm admittedly not very knowledgeable about the community style guidelines surrounding this sort of stuff.

Reply:
I don't think there's much consensus (not many people even do routine safety comments from what I've seen) but yeah I think it's a bit distracting in normal code comments